llm/gguf.go

```diff
@@ -316,7 +326,24 @@ func writeGGUFString(llm *gguf, w io.Writer, s string) error {
 	return err
 }
 
-func readGGUFV1Array(llm *gguf, r io.Reader) (a []any, err error) {
+type array struct {
+	size   int
+	values []any
+}
+
+func (a *array) MarshalJSON() ([]byte, error) {
+	if a.values == nil {
+		// TODO(bmizerany): Return null instead so clients can
+		// distinguish between an empty array and an array that was
+		// not collected.
+		return []byte("[]"), nil
+	}
+	return json.Marshal(a.values)
+}
+
+const maxArraySize = 1024
```

Comment:
Possible to add a comment here for future changes when we eventually increase it?

Reply:
This is now "configurable" and documented as such.